compiler/semantic/schema.go

```diff
@@ -460,6 +461,48 @@ func (s *subquerySchema) tableOnly(n ast.Node, name string, path []string) (sem.
 	return nil, fmt.Errorf("no such table %q", name) //XXX
 }
 
+func (a *aliasSchema) outColumns() ([]string, bool) {
+	return a.sch.outColumns()
+}
+
+func (d *dynamicSchema) outColumns() ([]string, bool) {
+	return nil, false
+}
+
+func (s *staticSchema) outColumns() ([]string, bool) {
+	return s.columns, true
+}
+
+func (s *selectSchema) outColumns() ([]string, bool) {
+	return s.out.outColumns()
+}
+
+func (j *joinSchema) outColumns() ([]string, bool) {
+	left, lok := j.left.outColumns()
+	right, rok := j.right.outColumns()
+	if !lok || !rok {
+		return nil, false
+	}
+	// Append all columns from the left except for those that overlap with
+	// the right as these will be overwritten. We/If we allow duplicate columns
+	// this should be changed to just concat both columns.
```

Comment:
"Append all columns from the left" doesn't describe accurately what's happening here.

```suggestion
	// Include all columns from the left except for those appearing on 
	// the right since that's what {...left,...right} does.
```